cdc/sink/producer/kafka/config.go

```diff
@@ -261,23 +257,36 @@ func newSaramaConfig(ctx context.Context, c *Config) (*sarama.Config, error) {
 		return nil, errors.Trace(err)
 	}
 	config.Version = version
-	// See: https://kafka.apache.org/documentation/#replication
-	// When one of the brokers in a Kafka cluster is down, the partition leaders
-	// in this broker is broken, Kafka will election a new partition leader and
-	// replication logs, this process will last from a few seconds to a few minutes.
-	// Kafka cluster will not provide a writing service in this process.
-	// Time out in one minute.
-	config.Metadata.Retry.Max = 120
-	config.Metadata.Retry.Backoff = 500 * time.Millisecond
-	// If it is not set, this means a metadata request against an unreachable
-	// cluster (all brokers are unreachable or unresponsive) can take up to
-	// `Net.[Dial|Read]Timeout * BrokerCount * (Metadata.Retry.Max + 1) +
-	// Metadata.Retry.Backoff * Metadata.Retry.Max`
-	// to fail.
-	// See: https://github.com/Shopify/sarama/issues/765
-	// and https://github.com/pingcap/tiflow/issues/3352.
+
+	// Producer fetch metadata from brokers frequently, if metadata cannot be
+	// refreshed easily, this would indicate the network condition between the
+	// capture server and kafka broker is not good.
+	// In the scenario that cannot get response from Kafka server, this default
+	// setting can help to get response more quickly.
+	config.Metadata.Retry.Max = 1
+	config.Metadata.Retry.Backoff = 100 * time.Millisecond
+	// This Timeout is useless if the `RefreshMetadata` time cost is less than it.
 	config.Metadata.Timeout = 1 * time.Minute
 
+	// Admin.Retry take effect on `ClusterAdmin` related operations,
+	// only `CreateTopic` for cdc now. set the `Timeout` to `1m` to make CI stable.
+	config.Admin.Retry.Max = 5
+	config.Admin.Retry.Backoff = 100 * time.Millisecond
+	config.Admin.Timeout = 1 * time.Minute
+
+	// Producer.Retry take effect when the producer try to send message to kafka
+	// brokers. If kafka cluster is healthy, just the default value should be enough.
+	// For kafka cluster with a bad network condition, producer should not try to
+	// waster too much time on sending a message, get response no matter success
+	// or fail as soon as possible is preferred.
+	config.Producer.Retry.Max = 3
+	config.Producer.Retry.Backoff = 100 * time.Millisecond
+
+	// make sure sarama producer flush messages as soon as possible.
+	config.Producer.Flush.Bytes = 0
+	config.Producer.Flush.Messages = 0
+	config.Producer.Flush.Frequency = time.Duration(0)
```

Comment:
What's the default values? Does the change affects performance?

Reply:
this is the default value, just make it explicitly, this guarantees the performance.